src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
         resultBuilder.addAct(
-            new ValueConfirmedAct(this, {
+            new ValueDisconfirmedAct(this, {
                 value: this.state.value,
-                renderedValue:
-                    this.state.value !== undefined ? this.props.valueRenderer(this.state.value, input) : '',
+                renderedValue: this.props.valueRenderer(this.state.value, input),
             }),
         );
+        const suggestValue = this.getAmbiguousPair(this.state.value, input);
+        if (suggestValue !== this.state.value) {
+            this.setValue(suggestValue);
+            this.addInitiativeAct(
+                new SuggestValueAct(this, {
+                    value: suggestValue,
+                    renderedValue: this.props.valueRenderer(suggestValue!, input),
+                }),
+                resultBuilder,
+            );
+        } else {
+            // Discard the stored input
+            this.clear();
+            this.addInitiativeAct(new RequestValueAct(this), resultBuilder);
+        }
     }
 
-    private isFeedbackUndefinedAndValueNotChangedWhenConfirmingValue(input: ControlInput): boolean {
+    private isFeedbackWithValue(input: ControlInput): any {
         try {
             okIf(InputUtil.isValueControlIntent(input, AmazonBuiltInSlotType.NUMBER));
             const { feedback, action, target, values } = unpackValueControlIntent(
                 (input.request as IntentRequest).intent,
             );
             const valueStr = values[0].slotValue;
-            okIf(
-                InputUtil.actionIsMatchOrUndefined(action, [
-                    ...this.props.interactionModel.actions.set,
-                    ...this.props.interactionModel.actions.change,
-                ]),
-            );
+            okIf(InputUtil.feedbackIsMatch(feedback, FEEDBACK_TYPES));
+            okIf(InputUtil.actionIsUndefined(action));
             okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-            okIf(InputUtil.feedbackIsUndefined(feedback));
             okIf(InputUtil.valueStrDefined(valueStr));
-            okIf(this.state.lastInitiative.actName === ConfirmValueAct.name);
-            okIf(this.state.value === Number.parseInt(valueStr, 10));
-            this.handleFunc = this.handleFeedbackYesAndValueUndefinedWhenConfirmingValue;
+            this.handleFunc = this.handleFeedbackWithValue;
             return true;
         } catch (e) {
             return falseIfGuardFailed(e);
         }
     }
 
-    private handleFeedbackYesAndValueUndefinedWhenConfirmingValue(
-        input: ControlInput,
-        resultBuilder: ControlResultBuilder,
-    ): void {
-        this.state.isValueConfirmed = true;
-        this.state.lastInitiative.actName = undefined;
+    private handleFeedbackWithValue(input: ControlInput, resultBuilder: ControlResultBuilder): void {
         resultBuilder.addAct(
```

Comment:
Currently this method can be invoked with an affirmation of correctness.  The resulting dialog may be wrong
```
A: Was that 10?
U: Yes, 10
A: Sorry. What value?
```

Perhaps we need isAffirmWithValue  and isDisaffirmWithValue

Reply:
Yep dialog changes with affirmations and disaffirmation, refactored to two separate built-in handlers
` isAffirmWithValue` and `isDisaffirmWithValue`